### ISSUE TITLE
Support lazy loading big files (especially when used as pager)

### ISSUE THREAD

**Issue by 3v1n0** [neovim/neovim#11283]
- `nvim --version`: 0.3.8
- `vim -u DEFAULTS` (version: ) behaves differently? No
- Operating system/version: Ubuntu 19.10
- Terminal name/version: Tilix
- `$TERM`: `xterm-256color`

I'm trying to use neovim as pager everywhere, unfortunately it's not possible when it comes to load big files or pipe some big output (such as a `git log -p`).

So for example, in a repository with various years and lots of commits, when I'm using `neovim` as pager  (even disabling all the plugins and configs), when piping to `more`, `less` or `tig` (in the case of git) the file is loaded lazily and only the first lines are shown while the others are loaded in background (in my test a log with 12187253 lines takes about 30 seconds to be read in `tig`).

This doesn't happen with (neo)vim, and it waits all the data to be completed before starting parsing it and then present it to the user. Making it impossible to use in such scenarios.

So, ideally when opening big files and when loading from stdin even vim should just fstream to the last-used position (in case of big files) or open the first lines while loading in the bg the rest.

I assume this won't make the syntax highlighting working properly for all the languages, but indeed it could be disabled until the end, or in the pager case, not strictly needed as the ansi codes could be used.

**Comment by justinmk** [comment]
Sounds great. We want this too. Someone needs to spend the time on it :)

https://github.com/neovim/neovim/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+pager

Duplicate of https://github.com/neovim/neovim/issues/614

See also https://github.com/neovim/neovim/issues/9462
